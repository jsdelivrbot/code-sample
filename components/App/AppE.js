import React from 'react'
import App from './App'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'

const createMockStore = configureMockStore()
const initialState = {}
const store = createMockStore(initialState)

const AppE = () => {
  return (
    <Provider store={store}>
      <div className='AppE'>
        <h1 className='exampleHeader'>Default</h1>

        <App>
          <div>children</div>
        </App>
      </div>
    </Provider>
  )
}

export default AppE
