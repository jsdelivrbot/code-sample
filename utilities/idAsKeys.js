import keyBy from 'lodash/keyBy'
import { array, checkPropTypes } from 'prop-types'

const propTypes = {
  items: array,
}

const idAsKeys = function ({ items }) {
  checkPropTypes(propTypes, arguments[0], 'prop', 'idAsKeys')

  return keyBy(items, (item) => {
    return item.data.id
  })
}

export default idAsKeys
