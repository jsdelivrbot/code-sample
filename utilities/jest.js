export const describe = global.describe

export const it = global.it

export const expect = global.expect

export const beforeEach = global.beforeEach

export const afterEach = global.afterEach
