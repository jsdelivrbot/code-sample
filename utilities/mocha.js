export const describe = window.describe

export const it = window.it

export default window.mocha
