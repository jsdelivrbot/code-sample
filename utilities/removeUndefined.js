import merge from 'lodash/merge'
import forIn from 'lodash/forIn'
import isObject from 'lodash/isObject'

// TODO: use destructor
function removeUndefined (o) {
  if (isObject(o) && !Array.isArray(o)) {
    let obj = merge({}, o)

    forIn(obj, function (val, key) {
      if (val === null || val === undefined) {
        delete obj[key]
      } else {
        obj[key] = removeUndefined(val)
      }
    })

    return obj
  }

  return o
}

export default removeUndefined
