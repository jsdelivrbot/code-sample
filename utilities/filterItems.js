import filter from 'lodash/filter'
import isMatch from 'lodash/isMatch'
import removeUndefined from 'utilities/removeUndefined'
import { object, checkPropTypes } from 'prop-types'

const propTypes = {
  items: object,
  filters: object,
}

const filterItems = function ({ items, filters }) {
  checkPropTypes(propTypes, arguments[0], 'prop', 'filterItems')

  const removedUndefinedFilters = removeUndefined(filters)

  return filter(items, ({ data }) => {
    return isMatch(data, removedUndefinedFilters)
  })
}

export default filterItems
