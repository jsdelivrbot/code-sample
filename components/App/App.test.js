import AppE from './AppE'
import React from 'react'
import renderer from 'react-test-renderer'
import { describe, it, expect } from 'utilities/jest'

describe('Test App', () => {
  it('should be able to render the default component', () => {
    const component = renderer.create(<AppE />)

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
