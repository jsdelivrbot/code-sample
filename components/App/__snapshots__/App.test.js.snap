// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App should be able to render the default component 1`] = `
<div
  className="AppE"
>
  <h1
    className="exampleHeader"
  >
    Default
  </h1>
  <div
    className="App"
  >
    <header>
      Twine Mail
    </header>
    <main>
      <div
        className="UnreadEmails"
      >
        <header>
          Unread Emails
        </header>
        <p
          className="NoItems"
        >
          There are no items available.
        </p>
        <ul />
      </div>
      <div
        className="ReadEmails"
      >
        <header>
          Read Emails
        </header>
        <p
          className="NoItems"
        >
          There are no items available.
        </p>
        <ul />
      </div>
    </main>
  </div>
</div>
`;
